src/main/java/codeu/model/data/User.java

```diff
@@ -23,36 +24,42 @@
   private final String name;
   private final String passwordHash;
   private final Instant creation;
+  private long messageCount;
+  private boolean isAdmin;
   private  String description;
 
-
   /**
    * Constructs a new User.
    *
-   * @param id the ID of this User
-   * @param name the username of this User
+   * @param id           the ID of this User
+   * @param name         the username of this User
    * @param passwordHash the password hash of this User
+   * @param creation     the creation time of this User
    * @param creation the creation time of this User
    * @param description of the username from profile section
    */
-  public User(UUID id, String name, String passwordHash, Instant creation, String description) {
+  public User(UUID id, String name, String passwordHash, Instant creation, boolean isAdmin) {
     this.id = id;
     this.name = name;
     this.passwordHash = passwordHash;
     this.creation = creation;
-    this.description = description;
+    this.messageCount = 0;
+    this.isAdmin = isAdmin;
+    this.description = "";
   }
 
-  /**
-   *
-   * @param id the ID of this User
-   * @param name the username of this User
-   * @param passwordHash the password hash of this User
-   * @param creation the creation time of this User
-   * @param "" for the user who is writing their description for first time
-   */
-  public User(UUID id, String name, String passwordHash, Instant creation) {
```

Comment:
So I dont think this constructor should be changed at all. This should be the base constructor for creating a User. When a user is created we can assume a few things about them like their messageCount (it should be zero at this point), we can assume their description (it should be ""), and  here we can computer their admin status (since we are using a static list of admins, we can just move that in here and set the admin status). With this being said we can keep the same parameters and just send the new parameters to the new constructor like so...
`this(id, name, passwordHash, creation, 0, computedIsAdmin, "")`

Overall, I am saying the top constructor (in line 41) is the one you want to change to include the new parameters because this is the one the PersistentDataStore needs, but this constructor should pretty much remain the same and just pass the default values to the other constructor.

Reply:
Done